aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -280,13 +424,23 @@ module aptos_token_objects::token {
 
         let Token {
             collection,
-            index,
+            index: deprecated_index,
             description: _,
             name: _,
             uri: _,
             mutation_events,
         } = move_from<Token>(addr);
 
+        let index = if (exists<TokenConcurrentFieldsAppendix>(addr)) {
```

Comment:
There seems to be a lot of duplication with 

if (exists<TokenConcurrentFieldsAppendix>(addr)) {
  borrow_global or borrow_global_mut
} else {
  ...
}
Can we make these into inline functions?

Reply:
unfortunatelly I don't see a way to make a function to wrap  that - it would need to be able to return different types (i.e. TokenConcurrentFieldsAppendix vs Token) , which I don't think is possible.